allennlp/modules/seq2seq_encoders/pytorch_seq2seq_wrapper.py

```diff
@@ -0,0 +1,38 @@
+import torch
+from torch.nn.utils.rnn import pad_packed_sequence, pack_padded_sequence
+
+from allennlp.common.tensor import sort_batch_by_length
+from allennlp.modules import Seq2SeqEncoder
+
+
+class PytorchSeq2SeqWrapper(Seq2SeqEncoder):
+    """
+    Pytorch's RNNs have two outputs: the hidden state for every time step, and the hidden state at
+    the last time step for every layer.  We just want the first one as a single output.  This
+    wrapper pulls out that output, and adds a :func:`get_output_dim` method, which is useful if you
+    want to, e.g., define a linear + softmax layer on top of this to get some distribution over a
+    set of labels.  The linear layer needs to know its input dimension before it is called, and you
+    can get that from ``get_output_dim``.
+    """
+    def __init__(self, module: torch.nn.modules.RNNBase) -> None:
+        super(PytorchSeq2SeqWrapper, self).__init__()
+        self._module = module
+
+    def get_output_dim(self) -> int:
+        return self._module.hidden_size * (2 if self._module.bidirectional else 1)
+
+    def forward(self,  # pylint: disable=arguments-differ
+                inputs: torch.Tensor,
+                sequence_lengths: torch.LongTensor = None) -> torch.Tensor:
+
+        if sequence_lengths is None:
```

Comment:
yes, or if you're doing something which doesn't require batching, like running an LSTM over a fixed size number of memory slots or something like they did in that Dynamic Memory Networks paper. I can remove this if you think it's confusing.